internal/providercache/installer.go

```diff
@@ -439,20 +442,31 @@ func (i *Installer) ensureProviderVersionsInstall(
 	authResults := map[addrs.Provider]*getproviders.PackageAuthenticationResult{} // record auth results for all successfully fetched providers
 
 	for provider, version := range need {
-		if err := ctx.Err(); err != nil {
+		traceCtx, span := tracing.Tracer().Start(ctx,
+			"Install Provider",
+			trace.WithAttributes(
+				attribute.String(tracing.ProviderAddressAttributeName, provider.String()),
+				attribute.String(tracing.ProviderVersionAttributeName, version.String()),
```

Comment:
I guess I reviewed these in the wrong order :facepalm: so I should really have left https://github.com/opentofu/opentofu/pull/2694#discussion_r2056333802 here instead.